### ISSUE TITLE
Add HOWTO: Circular Convs

### ISSUE THREAD

**Issue by marcvanzee** [google/flax#903]
XLA convs natively do not support this, but you can always manually concat a kernel/2 per dim amount of data on the edges to achieve it at some data layout cost.

Example code from @levskaya (in old API, but should be trivial to modify to Linen):

```
class CircularConv(nn.Module):
  def apply(self, inputs, features, kernel_size, **kwargs):
    padding = [(0, 0)] + [(k//2, k//2) for k in kernel_size] + [(0, 0)]
    inputs = jnp.pad(inputs, padding, mode='wrap')
    return nn.Conv(inputs, features, kernel_size, padding='VALID', **kwargs)
```

**Comment by VolodyaCO** [comment]
Will this be added as a `padding="CIRCULAR"` option as in [neural-tangents](https://neural-tangents.readthedocs.io/en/latest/_modules/neural_tangents/stax.html#Conv)?

**Comment by marcvanzee** [comment]
@VolodyaCO interesting suggestion! If you think this would be generally useful, could you create a feature request issue proposing this?

**Comment by VolodyaCO** [comment]
Hey @marcvanzee I just opened [it](https://github.com/google/flax/issues/971)

**Comment by marcvanzee** [comment]
Thanks! I'll lower the priority of this HOWTO since it may be possible we (or JAX) will actually add this as a feature.

**Comment by marcvanzee** [comment]
Closing this since @sgrigory implemented circular padding in #1661. Thanks @sgrigory!
